### PR TITLE
Thm Timings NOT READY FOR MERGE 

### PR DESCRIPTION
**Pull request by utunga**
NOT READY FOR MERGE 

just creating a PR so its easy to see the changes that have ended up a little bit 'orphaned' by the refactors that have taken place in Mozilla/DeepSpeech 0.5.1

### CODE REVIEW

File: native_client/modelstate.cc [hunk 1]
```diff
@@ -80,3 +80,105 @@ ModelState::decode_metadata(DecoderState* state)
...
+//   metadata->num_items = best.tokens.size();
+//   metadata->probability = best.probability;
+
+//   std::unique_ptr<MetadataItem[]> items(new MetadataItem[metadata->num_items]());
+
+//   // Loop through each character
+//   for (int i = 0; i < best.tokens.size(); ++i) {
+
+//     // best guess from acoustic model, for the timestep corresponding to chosen character
+//     int argmax = arg_max(&logits[best.timesteps[i] * num_classes], num_classes);
```
Comment (utunga) [inline native_client/modelstate.cc]: What makes this a bit tricky is that what they previously passed to this method <probs> could be interpreted as logits but now they pass <state> which doesn't contain probs, far as i can tell.

**Comment by utunga** [comment]
Oh no! Argh. So Sorry this was supposed to be PR to the http://github.com/Tehikumedia/deepspeech not the main Mozilla one. So far from ready. PLease ignore
